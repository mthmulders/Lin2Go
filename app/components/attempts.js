import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';

import styles from '../styles';

export const Attempts = (props) => {
  const { attempts } = props;

  const renderLetter = (letterWithScore, idx) => {
    let style = styles.attemptLetter;
    if (letterWithScore.score === 2) {
      style = styles.attemptLetterOnRightLocation;
    } else if (letterWithScore.score === 1) {
      style = styles.attemptLetterOnOtherLocation;
    }
    return (<Text style={ style } key={ idx }>{ letterWithScore.letter }</Text>)
  };

  const renderAttempt = (attempt, attemptId) => {
    const letters = attempt.word.split('');
    const lettersWithScore = letters.map((elem, index) => {
      return {
        letter: elem,
        score: attempt.score[index]
      }
    });
    return (
      <View style={ styles.attempt } key={ attemptId }>
        { lettersWithScore.map((elem, index) => renderLetter(elem, `${attemptId}.${index}`)) }
      </View>
    );
  };

  return (
    <View style={ styles.attempts }>
     { attempts.map(renderAttempt) }
    </View>
  );
};

const mapStateToProps = (state) => {
  return {
    attempts: state.game.attempts
  };
};

const mapDispatchToProps = (dispatch) => {
  return { };
};

Attempts.propTypes = {
  attempts: React.PropTypes.arrayOf(React.PropTypes.shape({
    word: React.PropTypes.string.isRequired
  })).isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(Attempts);
