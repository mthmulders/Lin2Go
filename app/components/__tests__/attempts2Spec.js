//
// Once https://github.com/facebook/react/issues/7386 is released, this
// file can be merged with attemptsSpec.js
//
jest.unmock('../attempts');

import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import { Attempts } from '../attempts';

describe('The attempts component', () => {

  it('should give visual feedback on the score of the attempt', () => {
    // Arrange
    const attempts = [
      {word: 'kiwis', score: [0, 0, 0, 0, 2]},
      {word: 'peren', score: [1, 1, 0, 2, 0]}
    ];

    // Act
    const tree = renderer.create(<Attempts attempts={ attempts } />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

});
