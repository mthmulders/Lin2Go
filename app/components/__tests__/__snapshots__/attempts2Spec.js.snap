exports[`The attempts component should give visual feedback on the score of the attempt 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center"
    }
  }>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "space-between"
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      k
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      i
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      w
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      i
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "red",
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      s
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "space-between"
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "yellow",
          "fontSize": 18,
          "justifyContent": "space-between",
          "overflow": "hidden",
          "textAlign": "center",
          "width": 25
        }
      }>
      p
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "yellow",
          "fontSize": 18,
          "justifyContent": "space-between",
          "overflow": "hidden",
          "textAlign": "center",
          "width": 25
        }
      }>
      e
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      r
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "red",
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      e
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "justifyContent": "space-between",
          "textAlign": "center",
          "width": 25
        }
      }>
      n
    </Text>
  </View>
</View>
`;
